### PR TITLE
Fixes the set accessor 

### PR DESCRIPTION
**Pull request by Anipik**
Related to https://github.com/dotnet/runtime/issues/33631
Related Runtime PR https://github.com/dotnet/runtime/pull/40296

### CODE REVIEW

**Comment by danmosemsft** [comment]
Link to issue?

**Comment by Anipik** [comment]
Added to Description

File: src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs [hunk 1]
```diff
@@ -122,9 +122,17 @@ public override SyntaxNode VisitConstructorDeclaration(ConstructorDeclarationSyn
             return node.WithBody(block);
         }
 
+        public override SyntaxNode VisitDestructorDeclaration(DestructorDeclarationSyntax node)
+        {
+            string message = "{ throw new System.PlatformNotSupportedException(" + $"{ _message }); " + " }\n";
```
Comment (eerhardt) [inline src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs:127]: (minor) this chunk of code is repeated 4 times now, can we refactor to a helper method?

File: src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs [hunk 2]
```diff
@@ -122,9 +122,17 @@ public override SyntaxNode VisitConstructorDeclaration(ConstructorDeclarationSyn
             return node.WithBody(block);
         }
 
+        public override SyntaxNode VisitDestructorDeclaration(DestructorDeclarationSyntax node)
+        {
```
Comment (eerhardt) [inline src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs:126]: The other 3 methods check for a null Body, but not this one. Is that intentional?

**Comment by Anipik** [comment]
> Have we tested this works end-to-end in dotnet/runtime?

Yes i tested it end to end, i found these edge cases while diffing the dlls.

File: src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs (hunk 2 above)
Comment (Anipik) [reply src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs:126]: The null check is for abstract members which dont provide implementation. Mainly used for interface members.

File: src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs (hunk 1 above)
Comment (Anipik) [reply src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs:127]: i can add a default message field, Its the same accept for properties

**Pull request merged**